src/koopman.jl

```diff
@@ -45,7 +45,7 @@ function koopman_expectation(g,u0s,ps,prob,args...;maxiters=0,
     # Define the integrand for expectation
     if batch == 0
       # create numerical state space values
-      ext_state_val = zeros(length(ext_state))
+      ext_state_val = vcat(zero.(eltype.(ext_state))...)
```

Comment:
I could use some suggestions on a better way to do this. Essentially the extended state space `ext_state` can have a mixture of types, e.g. `Float64`, `Distribution`, `Dual`... Need to create a zero vector the same length as extended states space, but need to convert Float types when AD used on subset of states. This works for ForwardDiff, but seems ugly.